src/main/java/org/opentripplanner/apis/transmodel/model/stop/QuayType.java

```diff
@@ -380,20 +380,21 @@ public static GraphQLObjectType create(
           .name("flexibleArea")
           .description("Geometry for flexible area.")
           .type(GeoJSONCoordinatesScalar.getGraphQGeoJSONCoordinatesScalar())
-          .dataFetcher(environment ->
-            (
-              environment.getSource() instanceof AreaStop areaStop
-                ? areaStop.getGeometry().getCoordinates()
-                : null
-            )
-          )
+          .dataFetcher(environment -> {
+            if (environment.getSource() instanceof AreaStop areaStop) {
+              return areaStop.getGeometry().getCoordinates();
+            } else if (environment.getSource() instanceof GroupStop groupStop) {
+              return groupStop.getEncompassingAreaGeometry().getCoordinates();
```

Comment:
What is returned when you call `getGeometry` on a `GroupStop`?

Reply:
The Geometries of the stops that are in the GroupStop

Reply:
I'm a bit hesitant about the `instanceOf` check. @t2gran are you on board with this?

Reply:
But Thomas is way stricter with these kinds of things, so if it's good enough for him, it's good enough for me.

Reply:
Discussed with Thomas this morning. I will do some refactoring to the StopLocation interface and the implementations to get rid of instanceof checks.

Regarding the discussion about the existing getGeometry method vs the getEncompassingAreaGeometry I introduce here to get the area of the GroupStop, I'll keep the new method and further make it a part of the StopLocation interface. This will let us implement it as null for RegularStop, and original defining areas for Area and GroupStop. We can't use the existing getGeometry method without also checking the type of stop, since RegularStop has a Geometry that is a point, and the API field is explicitly an area.

Reply:
Made the suggested changes.

Reply:
Resolving this since we're continuing the conversation in other comments.